Classes/BITCrashManager.m

```diff
@@ -116,12 +117,14 @@ static void bit_save_events_callback(siginfo_t __unused *info, ucontext_t __unus
   if (fd < 0) {
     return;
   }
-  
+
+  pthread_mutex_lock(&BITEventsStringMutex);
```

Comment:
❌ `pthread_mutex_lock` is [not async-signal-safe](http://www.manpages.info/macosx/sigaction.2.html), I'm not sure we should use it here

Reply:
How about using atomics instead?

Reply:
IMHO we should not worry about this callback right now as it will only ever be executed post-crash-time and we can by no means guarantee what happens at crash time. To make sure the callback operates on a "buffer" of events that is not corrupt, it needs to operate on it's own buffer (like we do in Mobile Center). While I think it'd be a good idea to implement this in HockeyApp, too, we should hold off for now and focus on making BITChannel threadsafe within itself.

Does that make sense? I'm working on a PR that includes ideas from a previous attempt to reproduce the issue.